### PR TITLE
ci: reduce API requests in download wheel job

### PR DESCRIPTION
**Pull request by P403n1x87**
We reduce the number of requests to the GitHub API made by the GitLab wheel download workflow to reduce the chances of going over the request rate limit. We also make sure that the job re-runs reactively when the GitHub workflow has completed.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: .gitlab/download-wheels-from-gh-actions.sh [hunk 1]
```diff
@@ -6,39 +6,51 @@ if [ -z "$CI_COMMIT_SHA" ]; then
...
 
 mkdir pywheels
 cd pywheels
 
-echo "Github workflow finished. Downloading wheels"
+if ! gh run download $RUN_ID --repo DataDog/dd-trace-py ; then
+  echo "Waiting for workflow to finish"
+
+  # Give time to the job to finish
+  sleep 600 # 10 minutes
```
Comment (romainkomorndatadog) [inline .gitlab/download-wheels-from-gh-actions.sh:44]: Maybe 10 minutes is a little bold (I know, I know, I suggested it). 🤔 

`sleep 180` would probably be pretty reasonable, all things considered?
```suggestion
  sleep 180 # 3 minutes
```

Comment (brettlangdon) [reply .gitlab/download-wheels-from-gh-actions.sh:44]: 👍🏻 if we do a sleep 180 above, then we might already be 1/3 done with the job by the time we hit this line.

180 here sounds good too, means we might waste 2-3 minutes sleeping and waiting, but "not too bad", we can always lower later too.

**Review by brettlangdon — APPROVED** [review]
agree with reducing the `sleep 600`, otherwise lgtm

File: .gitlab/download-wheels-from-gh-actions.sh (hunk 1 above)
Comment (P403n1x87) [inline .gitlab/download-wheels-from-gh-actions.sh:44]:
```suggestion
  sleep 300 # 5 minutes
```

**Review by romainkomorndatadog — APPROVED** [review]

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**